src/main/java/com/flipkart/zjsonpatch/JsonPatch.java

```diff
@@ -131,7 +131,7 @@ public static void validate(JsonNode patch) throws InvalidJsonPatchException {
     }
 
     public static JsonNode apply(JsonNode patch, JsonNode source, EnumSet<CompatibilityFlags> flags) throws JsonPatchApplicationException {
-        ApplyProcessor processor = new ApplyProcessor(source);
+        InPlaceApplyProcessor processor = new InPlaceApplyProcessor(source);
```

Comment:
Gopi, you're absolutely right. By mistake. Will fix and also add tests for `apply` doesn't mess with the source.

Thanks for the catch